### PR TITLE
Image display and About page

### PR DESCRIPTION
**Pull request by tsdobbs**
Two related features in this PR:

1. Added an internal "About" page
    - Added about_text.html and wrote description and links there
    - Moved slideshow and Youtube embed to about page.
2. Functionality for ad-hoc photo additions to the library - arbitrary process photos we want to display or, as in this case, art photos that we do not yet have uploaded to the artpieces table in the DB.
    - On render, generates a list of all photos in a defined directory, plus captions for them from EXIF information. The full list is added to the slideshow section in a random order
    - Captions are included with every photo, where available

For the sake of speed, the script references an internal directory and the current photos have been committed to the repo, but will probably eventually want to rewrite to use a URI set by an ENV var and get a media server.

### CODE REVIEW

**Comment by tsdobbs** [comment]
@brcb20 Obviously a quick manual test that it runs for you would be great, but also if you have design thoughts I'm open to them. My main goal is to push this by EOD tomorrow so that I can send out everyone's images via email finally and not be providing a dead link to the "/about" page.

File: web/api/user/utilities.py [hunk 1]
```diff
@@ -1,10 +1,35 @@
...
-    return image
\ No newline at end of file
+    return image
+
+"""
+Return a list of images in the 'gallery' folder and their descriptions
+Output is list of tuples (image_location, image_description)
+    output list is in random order for random display order every time
+"""
+def get_gallery_images():
```
Comment (brcb20) [inline web/api/user/utilities.py:19]:
```
def get_gallery_images():
    internal_path_prefix = './web'
    public_gallery_path = '/static/img/gallery/'

    image_paths = [
        public_gallery_path + filename
        for filename in os.listdir(internal_path_prefix + public_gallery_path)
    ]

    image_descriptions = list()
    for image_path in image_paths:
        with Image.open(internal_path_prefix + image_path) as image:
            # Exif ID 270 = ImageDescription
            image_descriptions.append(image.getexif().get(270))
            #now check aspect ratio and normalize

    image_metadata = list(zip(image_paths, image_descriptions))
    random.shuffle(image_metadata)

    return image_metadata
```

File: web/api/user/utilities.py [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+    internal_gallery_path = './web/static/img/gallery/'
+    public_gallery_path = '/static/img/gallery/'
+
+    imagename_list = [internal_gallery_path + filename for filename in os.listdir(internal_gallery_path)]
+
+    desc_list = list()
+    for img_name in imagename_list:
+        with Image.open(img_name) as image:
+            desc_list.append(image.getexif().get(270)) #Exif ID 270 = ImageDescription
+            #now check aspect ratio and normalize
```
Comment (brcb20) [inline web/api/user/utilities.py:29]: Can we manually size the images with the same aspect ratio before uploading to git? Preferably the same dimensions too.

File: web/templates/nav.html [hunk 3]
```diff
@@ -17,14 +17,14 @@
 
 	<div class="collapse navbar-collapse" id="navbarSupportedContent">
 		<ul class="navbar-nav align-items-center">
-			<li class="nav-item active ml-md-3">
-				<a class="nav-link" href="#">
-					Home<span class="sr-only">(current)</span>
+			<li class="nav-item ml-md-3">
```
Comment (brcb20) [inline web/templates/nav.html:20]: Ideally, we find a way to keep the active tag. Maybe we inject it into the nav.html template.

File: web/templates/nav.html [hunk 4]
```diff
@@ -17,14 +17,14 @@
...
+				<a class="nav-link" href="/">
+					Home
 				</a>
 			</li>
 			<li class="nav-item ml-md-3">
-				<a class="nav-link" target="_blank" rel="noopener"
-					href="https://endeneublog.wordpress.com/2019/10/02/artbot-takes-shape/">
+				<a class="nav-link" rel="noopener"
+					href="/about">
```
Comment (brcb20) [inline web/templates/nav.html:27]: In the future, we should make the website a SPA and just pull the body of the about (and cache) when this link is clicked.

File: web/templates/slideshow.html [hunk 5]
```diff
@@ -3,58 +3,20 @@
...
-					src="/static/img/a9.jpg"
-					class="d-block w-100"
-					alt="Tim holding one of our first tests, the smiley face agar plate, before putting it into the incubator.">
-			</div>
-			<div class="carousel-item">
-				<img
-					src="/static/img/a11.jpg"
-					class="d-block w-100"
-					alt="Close up of pipette tip in agar plate.">
+					src="{{ img[0] }}"
```
Comment (brcb20) [inline web/templates/slideshow.html:14]: Not sure why I didn't do this on the first pass... great refactor :+1:

**Review by brcb20 — CHANGES_REQUESTED** [review]

File: web/api/user/utilities.py (hunk 2 above)
Comment (tsdobbs) [reply web/api/user/utilities.py:29]: The dimension thing killed me - two different cameras used two different aspect ratios and resolutions. Developing a recommended standard would help, but I came to the same conclusion I think you're expressing here: don't handle it in the code, and certainly not at runtime.

I'll make a new issue for developing/checking a standard image ratio/resolution, and remove this comment.

File: web/api/user/utilities.py [hunk 6]
```diff
@@ -1,10 +1,35 @@
...
+    imagename_list = [public_gallery_path + filename for filename in os.listdir(internal_gallery_path)]
+    
+    img_list = list(zip(imagename_list,desc_list))
```
[the remark this answers is not in the document]
Comment (tsdobbs) [reply web/api/user/utilities.py:32]: Added Flask-Caching dependency, which added a lot of functionality - very straightforward: https://pythonhosted.org/Flask-Caching/
Cached the about page on a 1-minute cycle and the images on a 1-hour cycle.

File: web/templates/nav.html (hunk 3 above)
Comment (tsdobbs) [reply web/templates/nav.html:20]: Yeah, I guess it does help. Added it back by adding:
`home_tag=' active', about_tag=''` in the home render, and `home_tag='', about_tag=' active'` in the about render, then adding `{{ home_tag }}` and `{{ about_tag }}` into their respective places on the nav.html template.

I don't love the solution, but it gets us there.

**Pull request merged**